drivers/adc/ad7616/ad7616.c

```diff
@@ -38,20 +40,280 @@
+	uint8_t data_11 = AD7616_GET_BIT(data, 11);
+	uint8_t data_12 = AD7616_GET_BIT(data, 12);
+	uint8_t data_13 = AD7616_GET_BIT(data, 13);
+	uint8_t data_14 = AD7616_GET_BIT(data, 14);
+	uint8_t data_15 = AD7616_GET_BIT(data, 15);
+
+	uint8_t crc_0 = AD7616_GET_BIT(crc, 0);
+	uint8_t crc_1 = AD7616_GET_BIT(crc, 1);
+	uint8_t crc_2 = AD7616_GET_BIT(crc, 2);
+	uint8_t crc_3 = AD7616_GET_BIT(crc, 3);
+	uint8_t crc_4 = AD7616_GET_BIT(crc, 4);
+	uint8_t crc_5 = AD7616_GET_BIT(crc, 5);
+	uint8_t crc_6 = AD7616_GET_BIT(crc, 6);
+	uint8_t crc_7 = AD7616_GET_BIT(crc, 7);
+
+	crc_out = data_15 ^ data_13 ^ data_11 ^ data_7 ^ data_6 ^ data_5 ^
+		  crc_3 ^ crc_5 ^ crc_7;
+
+	crc_out = (crc_out << 1) | (data_14 ^ data_12 ^ data_10 ^ data_6 ^
+				    data_5 ^ data_4 ^ crc_2 ^ crc_4 ^ crc_6);
+
+	crc_out = (crc_out << 1) |
+		  (data_15 ^ data_13 ^ data_11 ^ data_9 ^ data_5 ^ data_4 ^
+		   data_3 ^ crc_1 ^ crc_3 ^ crc_5 ^ crc_7);
+
+	crc_out = (crc_out << 1) |
+		  (data_15 ^ data_14 ^ data_12 ^ data_10 ^ data_8 ^ data_4 ^
+		   data_3 ^ data_2 ^ crc_0 ^ crc_2 ^ crc_4 ^ crc_6 ^ crc_7);
+
+	crc_out = (crc_out << 1) |
+		  (data_14 ^ data_13 ^ data_11 ^ data_9 ^ data_7 ^ data_3 ^
+		   data_2 ^ data_1 ^ crc_1 ^ crc_3 ^ crc_5 ^ crc_6);
+
+	crc_out = (crc_out << 1) | (data_15 ^ data_13 ^ data_12 ^ data_10 ^
+				    data_8 ^ data_6 ^ data_2 ^ data_1 ^ data_0 ^
+				    crc_0 ^ crc_2 ^ crc_4 ^ crc_5 ^ crc_7);
+
+	crc_out = (crc_out << 1) |
+		  (data_15 ^ data_14 ^ data_13 ^ data_12 ^ data_9 ^ data_6 ^
+		   data_1 ^ data_0 ^ crc_1 ^ crc_4 ^ crc_5 ^ crc_6 ^ crc_7);
+
+	crc_out = (crc_out << 1) | (data_14 ^ data_12 ^ data_8 ^ data_7 ^
+				    data_6 ^ data_0 ^ crc_0 ^ crc_4 ^ crc_6);
+
+	return crc_out;
+}
+
+/**
+ * @brief Compute the CRC for multiples pairs of channel
+ * @param results - Array of conversion result
+ * @param samples - Number of conversion result
+ * @return The computed CRC
+ */
+static uint8_t crc_channel_pairs(struct ad7616_conversion_result *results,
+				 uint32_t samples)
+{
+	uint8_t crc = 0;
+	uint8_t crc1 = 0;
+
+	for (size_t i = 0; i < samples; i++) {
+		crc1 = crc_channel(results[i].channel_a, crc);
+		crc = crc_channel(results[i].channel_b, crc1);
+	}
+
+	return crc;
+}
+
+/**
+ * @brief Check if the recived CRC is correct
+ * @param results - Array of conversion result
+ * @param samples - Number of conversion result
+ * @param found_crc - CRC recived from the device
+ * @return 0 if the computed CRC is the same as the device, 1 otherwise
+ */
+static int32_t check_crc(struct ad7616_conversion_result *results,
+			 uint32_t samples, uint8_t found_crc)
+{
+	uint8_t computed_crc = crc_channel_pairs(results, samples);
+
+	return !(computed_crc == found_crc);
+}
+
+/**
+ * @brief Enable CRC
+ * @param dev - The device structure.
+ * @return 0 on success, an error code otherwise
+ */
+static int32_t ad7616_enable_crc(struct ad7616_dev *dev)
+{
+	int32_t ret = 0;
+
+	ret = ad7616_write_mask(dev, AD7616_REG_CONFIG, 0b11, 0b11);
```

Comment:
please use macros instead, similarly to what you are doing for the other configurations